homeassistant/components/switch/telnet.py

```diff
@@ -0,0 +1,154 @@
+"""
+Support for switch controlled using a telnet connection.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.telnet/
+"""
+import logging
+import telnetlib
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA,
+                                             ENTITY_ID_FORMAT)
+from homeassistant.const import (
+    CONF_RESOURCE, CONF_FRIENDLY_NAME, CONF_OPTIMISTIC, CONF_SWITCHES,
+    CONF_VALUE_TEMPLATE, CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_COMMAND_STATE,
+    CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 23
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Required(CONF_RESOURCE): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_COMMAND_OFF): cv.string,
+    vol.Required(CONF_COMMAND_ON): cv.string,
+    vol.Optional(CONF_COMMAND_STATE): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    vol.Optional(CONF_OPTIMISTIC): cv.boolean,
```

Comment:
Can we drop the optimistic configuration and just assume it's optimistic if there's no command state defined, or is there something I'm missing? For example, assumed_state can just be `return self._command_state is None`.